### PR TITLE
Minimum driving code with static message

### PR DESCRIPTION
**Pull request by forbesashli**
### Changelist 
Changed the vc ioc and added some extra files to utilize the telem proto messages. Gives others platforms to do integration tests for their projects

### Testing Done
-  testing on hardware and works

### Resolved Tickets
none

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -9,4 +9,9 @@
 	],
 	"cmake.configureOnEdit": false,
 	"cmake.configureOnOpen": false,
+	"files.associations": {
+		"hw_adc.h": "c",
+		"io_sbgellipse.h": "c",
+		"hw_uart.h": "c"
+	},
```
Comment (Lucien950) [inline .vscode/settings.json:16]:
```suggestion
```

File: firmware/dev/f4dev/cubemx/f4dev.ioc.md5
Comment (Lucien950) [inline firmware/dev/f4dev/cubemx/f4dev.ioc.md5:1]: please keep this

File: firmware/quadruna/VC/src/tasks.c [hunk 2]
```diff
@@ -419,7 +420,7 @@ void tasks_run1kHz(void)
     {
         const uint32_t start_time_ms = osKernelGetTickCount();
 
-        hw_watchdog_checkForTimeouts();
+        // hw_watchdog_checkForTimeouts();
```
Comment (Lucien950) [inline firmware/quadruna/VC/src/tasks.c:423]:
```suggestion
        hw_watchdog_checkForTimeouts();
```

File: firmware/quadruna/VC/src/tasks.c [hunk 3]
```diff
@@ -304,7 +304,7 @@ void tasks_init(void)
     HAL_ADC_Start_IT(&hadc3);
 
     // TODO: Re-enable watchdog (disabled because it can get annoying when bringing up a board).
-    hw_watchdog_init(hw_watchdogConfig_refresh, hw_watchdogConfig_timeoutCallback);
+    // hw_watchdog_init(hw_watchdogConfig_refresh, hw_watchdogConfig_timeoutCallback);
```
Comment (Lucien950) [inline firmware/quadruna/VC/src/tasks.c:307]:
```suggestion
    hw_watchdog_init(hw_watchdogConfig_refresh, hw_watchdogConfig_timeoutCallback);
```

**Review by Lucien950 — CHANGES_REQUESTED** [review]
quick fixes

File: firmware/quadruna/VC/src/app/states/app_allStates.c [hunk 4]
```diff
@@ -19,6 +20,11 @@
 #include "io_imu.h"
 
 #define IGNORE_HEARTBEAT_CYCLES 3U
+extern UART_HandleTypeDef huart1;
+static UART               modem900_uart = { .handle = &huart1 };
+
+extern UART_HandleTypeDef huart3;
+static UART               modem2G4_uart = { .handle = &huart3 };
```
Comment (Lucien950) [inline firmware/quadruna/VC/src/app/states/app_allStates.c:27]: please handle this in tasks.c

Comment (Lucien950) [reply firmware/quadruna/VC/src/app/states/app_allStates.c:27]: or make a mode wrapper in `io_telemMessage.h`

File: firmware/quadruna/VC/src/io/io_telemMessage.c [hunk 5]
```diff
@@ -0,0 +1,51 @@
+#include "io_telemMessage.h"
+#include "cmsis_os.h"
+#include "queue.h"
+#include "main.h"
+#include "../../../build_fw_deploy/firmware/quadruna/VC/telem.pb.h"
+#include "../../nanopb/pb_encode.h"
+#include "../../nanopb/pb_decode.h"
```
Comment (Lucien950) [inline firmware/quadruna/VC/src/io/io_telemMessage.c:7]: uh

**Review by Lucien950 — CHANGES_REQUESTED** [review]
harder things to fix

File: firmware/quadruna/VC/src/tasks.c [hunk 6]
```diff
@@ -397,6 +404,7 @@ void tasks_run100Hz(void)
 
         // Watchdog check-in must be the last function called before putting the
         // task to sleep.
+        io_telemMessage_broadcast(&modem900_uart);
```
Comment (Lucien950) [inline firmware/quadruna/VC/src/tasks.c:407]: nit, but the standard throughout the car is that you pass this type of metadata in the init function, then store it as a static variable inside `io_telemMessage.c`

**Review by Lucien950 — CHANGES_REQUESTED** [review]
1 more

File: firmware/quadruna/VC/src/io/io_telemMessage.h [hunk 7]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+#include "hw_uart.h"
+#include "../../../build_fw_deploy/firmware/quadruna/VC/telem.pb.h"
+#include "../../nanopb/pb_encode.h"
+#include "../../nanopb/pb_decode.h"
```
Comment (Lucien950) [inline firmware/quadruna/VC/src/io/io_telemMessage.h:5]: we should just strategically target include directories

File: firmware/quadruna/VC/src/io/io_telemMessage.c [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+#define QUEUE_BYTES 4 * QUEUE_SIZE // this is all temp
+
+static bool    proto_status;
+static uint8_t proto_msg_length;
+static uint8_t proto_buffer[QUEUE_SIZE]; // TODO: verify that this is the needed size (most likely can be smaller)
+TelemMessage   message = TelemMessage_init_zero;
+
+void io_telemMessage_init()
+{
+    Modem900 = true; // if false, then using the 2.4GHz, not implemented now bc no 2.4 yet
```
Comment (Lucien950) [inline firmware/quadruna/VC/src/io/io_telemMessage.c:20]: what

**Review by Lucien950 — CHANGES_REQUESTED** [review]

**Review by gtaharaedmonds — COMMENTED** [review]
LGTM
IMG_2809

**Comment by Lucien950** [comment]
W goat

**Review by Lucien950 — APPROVED** [review]
I am happy

**Pull request merged**